sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1396,14 +1337,125 @@ class CastSuite extends CastSuiteBase {
 
     checkEvaluation(cast("abcd", DecimalType(38, 1)), null)
   }
+
+  test("SPARK-22825 Cast array to string") {
```

Comment:
(nit and this is not related to this PR though...) `SPARK-22825` -> `SPARK-22825:`

Reply:
I would prefer keeping them unchanged in this PR